javagen/src/main/java/com/azure/autorest/template/ProtocolSampleTemplate.java

```diff
@@ -0,0 +1,99 @@
+package com.azure.autorest.template;
+
+import com.azure.autorest.extension.base.model.codemodel.Parameter;
+import com.azure.autorest.extension.base.model.codemodel.RequestParameterLocation;
+import com.azure.autorest.model.clientmodel.ClassType;
+import com.azure.autorest.model.clientmodel.ClientMethod;
+import com.azure.autorest.model.clientmodel.ClientMethodParameter;
+import com.azure.autorest.model.clientmodel.MethodGroupClient;
+import com.azure.autorest.model.clientmodel.ProxyMethodExample;
+import com.azure.autorest.model.javamodel.JavaFile;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ProtocolSampleTemplate {
+    private static final ProtocolSampleTemplate _instance = new ProtocolSampleTemplate();
+
+    protected ProtocolSampleTemplate() {}
+
+    public static ProtocolSampleTemplate getInstance() {
+        return _instance;
+    }
+
+    public void write(ClientMethod method, MethodGroupClient client, String builderName, String filename, ProxyMethodExample example, JavaFile javaFile) {
```

Comment:
Personally prefer have an object to wrap all these, so you can do a proper `IJavaTemplate`.

Reply:
Is sample and example same here?

Reply:
It is kind of terminology.

I usually use sample when there is multiple `ProxyMethodExample` within it. But it is just me. So you can use whatever name you wish.

BTW, also note that service and put anything into the `name` of `ProxyMethodExample`. It could contain `.`, space, etc.

Reply:
Done.